java/DEVELOPER.md

```diff
@@ -0,0 +1,269 @@
+    ```
+
+    2. Build in debug mode:
+
+    ```bash
+    ./gradlew :client:buildAll
+    ```
+
+    3. Build in release mode:
+
+    ```bash
+    ./gradlew :client:buildAllRelease
+    ```
+
+### Troubleshooting
+
+Some troubleshooting issues:
+
+- Failed to find cargo after rustup: gradlew may need to be restarted to recognize the new path. If clearing the gradle cache doesn't work, you may need to restart your machine.
+- If build fails because cargo build compiler fails, make sure submodules are updated using git submodule update.
+- If protobuf 26.0 or earlier is detected, upgrade to the latest protobuf release.
+
+
+### Test
+
+To run all tests, use the following command:
+
+```bash
+./gradlew test
+```
+
+To run a unit test, use the following command:
+
+```bash
+./gradlew :client:test
+```
+
+To run FFI tests between Java and Rust, use the following command:
+
+```bash
+./gradlew :client:testFfi
+```
+
+To run end-to-end tests, use the following command:
+
+```bash
+./gradlew :integTest:test
+```
+
+To run a single test, use the following command:
+```bash
+./gradlew :integTest:test --tests '*.functionLoad_and_functionList' --rerun
+```
+
+To run one class, use the following command:
+```bash
+./gradlew :client:test --tests 'TransactionTests' --rerun
+```
+
+### Generate files
+To (re)generate protobuf code, use the following command:
+
+```bash
+./gradlew protobuf
+```
+
+### Submodules
+
+After pulling new changes, ensure that you update the submodules by running the following command:
+
+```bash
+git submodule update
+```
+
+### Linters
+
+Development on the Java wrapper may involve changes in either the Java or Rust code. Each language has distinct linter tests that must be passed before committing changes.
+
+#### Language-specific Linters
+
+**Java:**
+
+-   Spotless
+
+#### Running the linters
+
+1. Spotless
+    ```bash
+    # Run from the `java` folder
+    ./gradlew :spotlessApply
+    ```
+2. Rust
+    ```bash
+    # Run from the `java` folder
+    rustup component add clippy rustfmt
+    cargo clippy --all-features --all-targets -- -D warnings
+    cargo fmt --manifest-path ./Cargo.toml --all
+    ```
+
+### Implementing a command
+
+- A node is an instance of a Redis server, and a redis cluster is composed of multiple nodes working in tandem.
+- The redis commands can either have a standalone or cluster implementation dependent on their specifications.
+- A cluster command will require a note to indicate a node will follow a specific routing.
+Refer to https://redis.io/docs/latest/operate/oss_and_stack/reference/cluster-spec for more details on how hash slots work for cluster commands.
+
+When starting a command, check the redis_request.proto and request_type.rs to see whether the command has already been implemented in another language such as Python or Node.js.
+
+The BaseClient.java (standalone and cluster) will implement methods from the following interfaces listed in the java/client/src/main/java/glide/api/commands.
```

Comment:
```suggestion
Standalone and cluster clients both extend `BaseClient.java` and implement methods from the interfaces listed in `java/client/src/main/java/glide/api/commands`.
```
Add clickable links to the dir, `BaseClient.java`, Standalone and cluster

Reply:
Please re-read all the text ignoring all you knowledge that you already have.
Only a person who never dealt with the client will read this - it should be clear and complete.